src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,105 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private static final int NULL_INDEX = 0;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size > DEFAULT_LOAD_FACTOR * table.length) {
+            resize();
+        }
+        Node<K, V> checkNode = table[hashCode(key)];
+        Node<K, V> newNode = new Node<>(hashCode(key), key, value, null);
+        if (checkNode == null) {
+            size++;
+            table[hashCode(key)] = newNode;
+            return;
+        }
+        for (; ; ) {
+            if (key == checkNode.key) {
+                checkNode.value = value;
+                return;
+            }
+            if (checkNode.key != null && checkNode.key.equals(key)) {
+                checkNode.value = value;
+                return;
+            }
```

Comment:
```suggestion
            if (Objects.equals(key, checkNode.key)) {
                checkNode.value = value;
                return;
            }
```
you can already use Objects